packages/@aws-cdk/aws-sagemaker/README.md

```diff
@@ -195,3 +195,60 @@ const endpointConfig = new sagemaker.EndpointConfig(this, 'EndpointConfig', {
   ]
 });
 ```
+
+### Endpoint
+
+When you create an endpoint from an `EndpointConfig`, Amazon SageMaker launches the ML compute
+instances and deploys the model or models as specified in the configuration. To get inferences from
+the model, client applications send requests to the Amazon SageMaker Runtime HTTPS endpoint. For
+more information about the API, see the
+[InvokeEndpoint](https://docs.aws.amazon.com/sagemaker/latest/dg/API_runtime_InvokeEndpoint.html)
+API. Defining an endpoint requires at minimum the associated endpoint configuration:
+
+```typescript
+import * as sagemaker from '@aws-cdk/aws-sagemaker';
+
+declare const endpointConfig: sagemaker.EndpointConfig;
+
+const endpoint = new sagemaker.Endpoint(this, 'Endpoint', { endpointConfig });
```

Comment:
Instead of supplying an endpoint config construct, could you simply supply a raw config into `Endpoint`, and get `Endpoint` to spin up the `EndpointConfig` for you? What we're offering makes sense if you understand the use case of repeating endpointConfigs elsewhere, but may seem strange for someone who just wants one configuration on the endpoint. In that use case, it makes more sense to supply the raw config to `Endpoint` and then abstract away the `EndpointConfig` construct from the user entirely.

Maybe the resolution here is to add this as an issue and track +1s. If people want it, we can add it in later.

Reply:
> Instead of supplying an endpoint config construct, could you simply supply **a raw config** into `Endpoint`, and get `Endpoint` to spin up the `EndpointConfig` for you?

1. What do you mean by "_a raw config_"?
    1. Do you mean take each of [these members of `EndpointConfigProps`](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-sagemaker/lib/endpoint-config.ts#L132-L145) and add them to `EndpointProps` as optional arguments (while making the `IEndpointConfig` prop value optional as well)?
    1. Or possibly, take `EndpointConfigProps` as a parameter on `EndpointProps`?
1. In this scenario, would it also make sense to include an `addInstanceProductionVariant` on `Endpoint` so that customers can mutate the backing `EndpointConfig` in the same manner as they could using their own instance (this does have the added bonus of returning an `IEndpointInstanceProductionVariant` instance from the method, eliminating the need for customers to look up the variant by name).
    1. If so, should this mutation only be permitted when creating the `EndpointConfig` on behalf of customers, or should it also be supported in cases when they provide their own `EndpointConfig` instance?

Reply:
I am tossing 1i. out as the suggestion, along with ensuring that the "raw config" and the `IEndpointConfig` prop are mutually exclusive. 

However, this is one of those things that is easier to add and harder to take away. So I think this should be a no-op for now, and we wait to see if something like this is a common complaint amongst users.

Reply:
Sounds good. Once this PR gets merged, would you like me to proactively create an issue to track that feature request?

Reply:
Yep, that would be great :)

Reply:
Closing the loop on this thread, I've created the following issue to track interest in this feature:
* #23162